scripts/import-open-api.ts

```diff
@@ -0,0 +1,236 @@
+    case "binary":
+    case "date":
+    case "dateTime":
+    case "password":
+      return "string";
+
+    default:
+      return "any";
+  }
+};
+
+/**
+ * Return the output type from the $ref
+ *
+ * @param $ref
+ */
+const getRef = ($ref: ReferenceObject["$ref"]) => {
+  if ($ref.startsWith("#/components/schemas")) {
+    return pascal($ref.replace("#/components/schemas/", ""));
+  } else {
+    throw new Error("This library only resolve $ref that are include into `#/components/schemas` for now");
+  }
+};
+
+/**
+ * Return the output type from an array
+ *
+ * @param item item with type === "array"
+ */
+const getArray = (item: SchemaObject): string => {
+  const items = item.items!;
+  if (isReference(items)) {
+    return `${getRef(items.$ref)}[]`;
+  } else {
+    return `${getScalar(items)}[]`;
+  }
+};
+
+/**
+ * Return the output type from an object
+ *
+ * @param item item with type === "object"
+ */
+const getObject = (item: SchemaObject): string => {
+  if (item.additionalProperties) {
+    if (isReference(item.additionalProperties)) {
+      return `{[key: string]: ${getRef(item.additionalProperties.$ref)}`;
+    } else if (item.additionalProperties.oneOf) {
+      return `{[key: string]: ${item.additionalProperties.oneOf
+        .map(prop => (isReference(prop) ? getRef(prop.$ref) : getScalar(prop)))
+        .join(" | ")}}`;
+    }
+  }
+
+  return "any"; // fallback
+};
+
+/**
+ * Extract responses types from open-api specs
+ *
+ * @todo remove potential duplicates
+ * @param responses reponses object from open-api specs
+ * @param componentName name of the current component
+ */
+const getResponseTypes = (responses: Array<[string, ResponseObject | ReferenceObject]>, componentName: string) =>
+  responses
+    .map(([statusCode, res]) => {
+      if (isReference(res)) {
+        throw new Error("$ref are not implemented inside responses");
+      } else {
+        if (res.content && res.content["application/json"]) {
+          const schema = res.content["application/json"].schema!;
+          return isReference(schema) ? getRef(schema.$ref) : getScalar(schema);
+        } else if (res.content && res.content["application/octet-stream"]) {
+          const schema = res.content["application/octet-stream"].schema!;
+          return isReference(schema) ? getRef(schema.$ref) : getScalar(schema);
+        } else {
+          return "void";
+        }
+      }
+    })
+    .join(` | `);
+
+/**
+ * Import and parse the openapi spec from a yaml/json
+ *
+ * @param path abosulte path of the yaml/json openapi 3.0.x file
+ */
+const importSpecs = (path: string): OpenAPIObject => {
+  const data = readFileSync(path, "utf-8");
+  const { ext } = parse(path);
+  return ext === ".yaml" ? YAML.parse(data) : JSON.parse(data);
+};
+
+/**
+ * Generate a restful-react Get compoment from openapi operation specs
+ *
+ * @param operation
+ */
+const generateGetComponent = (operation: OperationObject, verb: string, route: string) => {
```

Comment:
file is called `import-open-api` and `generateGetComponent` consumes it. so consumer shouldn't be declared here. am I wrong?

Reply:
The entry point is the `main` function on bottom, and yes I will split this file in smaller, and also add a lot of unit tests! It was just more conveniant like this to iterate :wink: (it's a wip branch)